cpp/src/table/legacy/device_table_row_operators.cuh

```diff
@@ -187,6 +224,46 @@ struct typed_row_inequality_comparator
         return State::False;
     }
   }
+  template <typename ColType>
+  __device__
+    typename std::enable_if_t<std::is_floating_point<ColType>::value, State>
+      operator()(gdf_index_type lhs_row, gdf_index_type rhs_row,
+                 gdf_column const *lhs_column, gdf_column const *rhs_column,
+                 bool nulls_are_smallest)
+  {
+    const ColType lhs_data = static_cast<const ColType *>(lhs_column->data)[lhs_row];
+    const ColType rhs_data = static_cast<const ColType *>(rhs_column->data)[rhs_row];
+
+    bool isValid1, isValid2;
+    if (nullable)
+    {
+      isValid1 = gdf_is_valid(lhs_column->valid, lhs_row);
+      isValid2 = gdf_is_valid(rhs_column->valid, rhs_row);
+    }
+    else
```

Comment:
Is this logic correct? You only check for NaNs if it's not nullable? What if you have a nullable column with non-null NaN elements?

I think we need to decide whether NaN == NULL is true or false.

Reply:
That should absolutely be false.

Reply:
@harrism Yes. I figured later that I missed this case. I will fix it now.
`NaN == NULL` is `false` - Is this documented somewhere?